presto-main/src/main/java/io/prestosql/execution/QueryStateTimer.java

```diff
@@ -140,6 +140,10 @@ private void endQuery(long now)
         finishingTime.compareAndSet(null, nanosSince(beginFinishingNanos, now));
         executionTime.compareAndSet(null, nanosSince(beginPlanningNanos, now));
         endNanos.compareAndSet(null, now);
+
+        // Analysis is run in separate thread and there is no query state for analysis.
+        // In case when analysis thread was canceled the analysis should be marked as finished.
+        endAnalysis(now);
```

Comment:
why not do this even before `beginFinishing(now);` ?

Reply:
does it matter? I have no preference